packages/admin-panel-server/src/__tests__/viz-builder/dashboardVisualisation/DashboardVisualisationExtractor.test.ts

```diff
@@ -0,0 +1,417 @@
+import { DashboardVisualisationExtractor } from '../../../viz-builder/dashboardVisualisation/DashboardVisualisationExtractor';
+
+describe('DashboardVisualisationExtractor', () => {
+  describe('validation', () => {
+    it('throws error if viz does not have data field', () => {
+      expect(
+        () => new DashboardVisualisationExtractor({ presentation: {} }, yup.object(), yup.object()),
+      ).toThrow('data is a required field');
+    });
+
+    it('throws error if viz does not have presentation field', () => {
+      expect(
+        () => new DashboardVisualisationExtractor({ data: {} }, yup.object(), yup.object()),
+      ).toThrow('presentation is a required field');
+    });
+
+    describe('getDraftReport', () => {
+      it('throws error if viz does not have data.fetch', () => {
+        expect(() =>
+          new DashboardVisualisationExtractor(
+            { code: 'viz', data: { transform: [] }, presentation: {} },
+            yup.object(),
+            draftReportValidator,
+          ).getReport(),
+        ).toThrow('fetch is a required field');
+      });
+
+      it('throws error if viz does not have data.fetch.dataElements or data.fetch.dataGroups', () => {
+        expect(() =>
+          new DashboardVisualisationExtractor(
+            { code: 'viz', data: { fetch: {}, transform: [] }, presentation: {} },
+            yup.object(),
+            draftReportValidator,
+          ).getReport(),
+        ).toThrow('Requires "dataGroups" or "dataElements"');
+      });
+
+      it('throws error if viz does not have data.transform', () => {
+        expect(() =>
+          new DashboardVisualisationExtractor(
+            { code: 'viz', data: { fetch: { dataElements: ['BCD1'] } }, presentation: {} },
+            yup.object(),
+            draftReportValidator,
+          ).getReport(),
+        ).toThrow('transform is a required field');
+      });
+
+      it('throws error if viz does not have code', () => {
+        expect(() =>
+          new DashboardVisualisationExtractor(
+            {
+              data: { fetch: { dataElements: ['BCD1'] }, transform: [] },
+              presentation: {},
+            },
+            yup.object(),
+            draftReportValidator,
+          ).getReport(),
+        ).toThrow('Requires "code" for the visualisation');
+      });
+
+      it('can get a draft report', () => {
+        expect(
+          new DashboardVisualisationExtractor(
+            {
+              code: 'viz',
+              data: { fetch: { dataElements: ['BCD1'] }, transform: [] },
+              presentation: {},
+            },
+            yup.object(),
+            draftReportValidator,
+          ).getReport(),
+        ).toEqual({
+          code: 'viz',
+          config: {
+            fetch: {
+              dataElements: ['BCD1'],
+            },
+            transform: [],
+          },
+        });
+      });
+    });
+
+    describe('getLegacyReport', () => {
```

Comment:
Camel case in `getLegacyReport` slightly implies that this is a function (or a property) while this is not the case. Another approach would be using something like the following:

```suggestion
    describe('getReport() - legacy report', () => {
```

I like using `functionName()` to denote that the scope of a test block is this specific function. However, this is a personal preference and not something we have decided as a team